packages/edge-config/src/create-create-client.ts

```diff
@@ -92,6 +97,48 @@ export function createCreateClient({
         process.env.NODE_ENV === 'development' &&
         process.env.EDGE_CONFIG_DISABLE_DEVELOPMENT_SWR !== '1';
 
+      /**
+       * The edge config bundled at build time
+       */
+      const bundledEdgeConfig: BundledEdgeConfig | null =
+        connection && connection.type === 'vercel'
+          ? readBundledEdgeConfig(connection.id)
+          : null;
+
+      const isBuildStep =
+        process.env.CI === '1' ||
+        process.env.NEXT_PHASE === 'phase-production-build';
+
+      /**
+       * Ensures that the provided function runs within a specified timeout.
+       * If the timeout is reached before the function completes, it returns the fallback.
+       */
+      async function timeout<T>(
+        method: string,
+        key: string | string[] | undefined,
+        localOptions: EdgeConfigFunctionsOptions | undefined,
+        run: () => Promise<T>,
+      ): Promise<T> {
+        const ms = localOptions?.timeoutMs ?? timeoutMs;
+
+        if (typeof ms !== 'number') return run();
+
+        let timer: NodeJS.Timeout | undefined;
+        // ensure we don't throw within race to avoid throwing after run() completes
+        const result = await Promise.race([
+          new Promise<TimeoutError>((resolve) => {
+            timer = setTimeout(
+              () => resolve(new TimeoutError(edgeConfigId, method, key)),
+              ms,
+            );
+          }),
+          run(),
+        ]);
+        if (result instanceof TimeoutError) throw result;
+        clearTimeout(timer);
```

Comment:
```suggestion
        const result = await Promise.race([
          new Promise<TimeoutError>((resolve) => {
            timer = setTimeout(
              () => resolve(new TimeoutError(edgeConfigId, method, key)),
              ms,
            );
          }),
          run(),
        ]).finally(() => clearTimeout(timer));
        if (result instanceof TimeoutError) throw result;
```

Makes sure the timeout is always cleared when the race is done, otherwise someone could set a really high timeout value and keep the timer running, when the `run` function throws, preventing node from exiting.

Reply:
Great catch, fixed in 03bad5b5162cbd117863760954953c293039460f